packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -166,3 +186,19 @@ function assertPostpone() {
     )
   }
 }
+
+/**
+ * This is a bit of a hack to allow us to abort a render using a Postpone instance instead of an Error which changes React's
+ * abort semantics slightly.
+ */
+export function createPostponedAbortSignal(reason: string): AbortSignal {
```

Comment:
When would I want to use this function instead of `postponeWithTracking`? Or is there really only one spot where we'd want to use this and that's the stream options we pass to the static renderer? I ask mostly because I'm wondering if it should be inlined rather than as a util

Reply:
The technique of aborting a render with a postpone probably won't happen elsewhere because it only really makes sense when you are trying to turn a prerender into a fully static result.

The abort signal is really only intended to be used when aborting a render whereas the postponeWithTracking is to create a hole in the current render. I could move it into app-render but i'd have to duplicate the postpone assertion logic and it sort of colocates all the postpone interfaces in a single place. It's probably not the right long term factoring but I don't know if moving it to app-render is either so I want to leave it here for now